File: quick_sort.py
def quick_sort(arr):
  if len(arr) <= 1:
    return arr

  pivot = arr[0]
  mid = []
  left = []
  right = []

  for item in arr:
    if(item > pivot):
      right.append(item)
    elif (item < pivot):
      left.append(item)
    elif (item == pivot):
      mid.append(item)

  return quick_sort(left) + mid + quick_sort(right)
  
print(quick_sort([5, 10, -4, 321, 10, 23, 3]))

File: quickSort.js
var quickSort = function(arr) {
  if (arr.length <= 1) {
    return arr;
  }

  var pivot = arr[0];
  var mid = [];
  var left = [];
  var right = [];

  for(let i = 0; i < arr.length; i++) {
    if(arr[i] > pivot) {
      right.push(arr[i]);
    } else if(arr[i] < pivot) {
      left.push(arr[i]);
    } else if(arr[i] === pivot) {
      mid.push(arr[i])
    }
  }
  
  return quickSort(left).concat(mid, quickSort(right));
}

console.log(quickSort([5, 10, -4, 321, 10, 23, 3]));